invenio_accounts/domains.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2023 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Domain blocking listener."""
+
+from .models import DomainStatus
+
+
+def on_user_confirmed(app, user):
+    """Listener for when a user is confirmed."""
+    security = app.extensions["security"]
+    datastore = security.datastore
+
+    # Domain is inserted on domain list when a user confirms their email.
+    domain = datastore.find_domain(user.domain)
+    if domain is None:
+        domain = datastore.create_domain(user.domain)
```

Comment:
Given that we are doing a session.add() here, is a `db.session.commit` always happening after this?

Reply:
Yep, checked. ``user_confirmed`` signal is sent from the ``confirm_user()`` function that's called here, which will do the commit:

https://github.com/inveniosoftware/flask-security-fork/blob/b99556579f4bbc9d1425e79d1f9f0d115e5baf6a/flask_security/views.py#L146-L147